macos/FluentUI/Colors.swift

```diff
@@ -0,0 +1,304 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+// MARK: Colors
+
+@objc(MSFColors)
+public final class Colors: NSObject {
+	// MARK: - MSFColorPalette
+	
+	/// colors defined in asset catalog
+	@objc(MSFColorPalette)
+	public enum Palette: Int {
+		case pinkRed10
+		case red20
+		case red10
+		case orange30
+		case orange20
+		case orangeYellow20
+		case green20
+		case green10
+		case cyan30
+		case cyan20
+		case cyanBlue20
+		case cyanBlue10
+		case blue10
+		case blueMagenta30
+		case blueMagenta20
+		case magenta20
+		case magenta10
+		case magentaPink10
+		case gray40
+		case gray30
+		case gray20
+		case gray25
+		case gray50
+		case gray100
+		case gray200
+		case gray300
+		case gray400
+		case gray500
+		case gray600
+		case gray700
+		case gray800
+		case gray900
+		case gray950
+		case communicationBlue
+		case communicationBlueTint40
+		case communicationBlueTint30
+		case communicationBlueTint20
+		case communicationBlueTint10
+		case communicationBlueShade30
+		case communicationBlueShade20
+		case communicationBlueShade10
+		case dangerPrimary
+		case dangerTint40
+		case dangerTint30
+		case dangerTint20
+		case dangerTint10
+		case dangerShade30
+		case dangerShade20
+		case dangerShade10
+		case warningPrimary
+		case warningTint40
+		case warningTint30
+		case warningTint20
+		case warningTint10
+		case warningShade30
+		case warningShade20
+		case warningShade10
+		case successPrimary
+		case successTint40
+		case successTint30
+		case successTint20
+		case successTint10
+		case successShade30
+		case successShade20
+		case successShade10
+		case presenceAvailable
+		case presenceAway
+		case presenceBlocked
+		case presenceBusy
+		case presenceDnd
+		case presenceOffline
+		case presenceOof
+		case presenceUnknown
+		
+		public var color: NSColor {
+			if let fluentColor = NSColor(named: "FluentColors/" + self.name, bundle: FluentUIResources.resourceBundle) {
+				return fluentColor
+			} else {
+				preconditionFailure("invalid fluent color")
+			}
+		}
+		
+		public var name: String {
```

Comment:
why does name need to be public?

Reply:
The test app needs it, though it's unlikely other clients would need it.

Reply:
For app size every bit counts, can we just get the name using Mirrors in the test app so there's no impact to product code? https://developer.apple.com/documentation/swift/mirror